### ISSUE TITLE
NFT image doesn't show up

### ISSUE THREAD

**Issue by leobel** [Berry-Pool/nami-wallet#187]
I've minted a NFT using an address from Nami Wallet but token image doesn't show up. Here is the tx on testnet: https://explorer.cardano-testnet.iohkdev.io/en/transaction?id=4c8c7a8fa2f7ab013e13d523800b76bb12c93868dfc676d7e29ac3efb6edc14d where you can see the metadata

```
{
  "version": "1.0",
  "c053203b3daf86b6588640436792dc5fc156f5b964b0c633a8f38c95": {
    "TANGOCRAZY01": {
      "body": "Rounded",
      "eyes": "<eyes>",
      "face": "<face>",
      "name": "Tangocrypto TANGOCRAZY01",
      "pepe": "grillo",
      "color": "Blue",
      "image": "ipfs://QmaQPBSXuRRcJkCMV9vbJ2gX9wwX2mye7GsSEbQjZnp6Rs",
      "artist": "Tangocrypto",
      "rarity": "1",
      "twitter": "https://twitter.com/tango_crypto",
      "accessory": "None",
      "copyright": "Tangocrypto 2021",
      "mediaType": "png",
      "collection": "Tangocrypto Collection",
      "description": "Description for TANGOCRAZY01"
    }
  }
}
```

I've tried it on ccvault wallet (I just send the token) 
![Screen Shot 2022-01-05 at 6 45 54 PM](https://user-images.githubusercontent.com/16786232/148294270-430a6b68-8d95-48c4-8dc7-1ce3a8b74ab3.png)

**Comment by leobel** [comment]
After some further tests I think the issue is Nami wallet isn't showing the image if the `asset_name` is different from the `name` . I don't know if this is a bug or a business decision from Nami

**Comment by alessandrokonrad** [comment]
The actual assetName is `54616e676f63727970746f2054414e474f4352415a593031`, but you use in the metadata the assetName `TANGOCRAZY01`. These names have to match. Actually blockfrost is resolving that and nami just fetches token metadata from there.

**Comment by leobel** [comment]
Hi @alessandrokonrad when you say: "These names have to match" you mean the asset_name and the inner name right? I think it doesn't matter if they are UTF-8 or Hex encoded, they just need to match. If I use `TANGOCRAZY01` in both it works. At the end it seems a bug since I don't see any reason why they must match did you?

**Comment by alessandrokonrad** [comment]
> "These names have to match" you mean the asset_name and the inner name right? 

Yes exactly. They need to have the same encoding, however the problem is that in CIP-0025 I never specified what encoding should be used for the the inner assetName. Most projects went with utf-8 per default. Hex would be more universal of course, but blockfrost is converting the assetName to utf-8 and looks up the assetName in the metadata map.

**Comment by tinglycraniumplacidly** [comment]
Hi. We've tried uploading cNFT collections with metadata where the asset_name = name, and also where it doesn't. The metadata and/or the gif images themselves don't seem to make any difference.

I believe the issue is with the ipfs mirror that nami uses. For example,
https://ipfs.blockfrost.dev/ipfs/QmWNiyZ3fE4UPGCnxyspkr4n46xjwWMdEkvUkeAhMiMPAa
504's

whereas these
https://gw3.easy-ipfs.com/ipfs/QmWNiyZ3fE4UPGCnxyspkr4n46xjwWMdEkvUkeAhMiMPAa
https://ipfs.io/ipfs//QmWNiyZ3fE4UPGCnxyspkr4n46xjwWMdEkvUkeAhMiMPAa
work

We noticed that about half the assets in a collection will 504 with the blockfrost.dev links. The other two mirrors work 100% of the time without fail.
